### ISSUE TITLE
[sql/mysql/Positive-Technologies] test sample sql with root rule

### ISSUE THREAD

**Issue by Lson-L** [antlr/grammars-v4#2657]
I test root rule with sample sql,why the tree is nothing?

<img width="960" alt="b1400270f69ef70f2f593ca2e266a9e" src="https://user-images.githubusercontent.com/77139715/173788460-9dfcf066-5241-4b08-8d1a-81e99c9f806a.png">

**Comment by kaby76** [comment]
Open MySqlParser.g4, then see if there is a red underscore error under the name of the tokenVocab option. Right-click and in the pop-up, try "Generate ANTLR Recognizer" and see if that fixes the red underscore error. If so, try "test rule node" again.

**Comment by bkiers** [comment]
Don't you need a custom char stream, because the lexer expects upper case keywords?

From: https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Positive-Technologies 

> As SQL grammar are normally not case sensitive but this grammar implementation is, you must use a custom [character stream](https://github.com/antlr/antlr4/blob/master/runtime/Java/src/org/antlr/v4/runtime/CharStream.java) that converts all characters to uppercase before sending them to the lexer.

**Comment by Lson-L** [comment]
yes,i need a custom char stream,Thank you very much




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "antlr/grammars-v4"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2022年6月16日(星期四) 凌晨3:17
***@***.***&gt;;
***@***.******@***.***&gt;;
主题:&nbsp;Re: [antlr/grammars-v4] [sql/mysql/Positive-Technologies] test sample sql with root rule (Issue #2657)





 
Don't you need a custom char stream, because the lexer expects upper case keywords?
 
From: https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Positive-Technologies
  
As SQL grammar are normally not case sensitive but this grammar implementation is, you must use a custom character stream that converts all characters to uppercase before sending them to the lexer.

**Comment by kaby76** [comment]
>> Don't you need a custom char stream, because the lexer expects upper case keywords?

> yes,i need a custom char stream,Thank you very much

Folks, this doesn't make sense.

[MySqlLexer.g4](https://github.com/antlr/grammars-v4/blob/041b3afc6c7689324ed137834f4549b23a52ecd1/sql/mysql/Positive-Technologies/MySqlLexer.g4) (with [MySqlParser.g4](https://github.com/antlr/grammars-v4/blob/041b3afc6c7689324ed137834f4549b23a52ecd1/sql/mysql/Positive-Technologies/MySqlParser.g4)) was [updated a month ago](https://github.com/antlr/grammars-v4/commit/c810ec33085922e2138efd1b3268c481997f9573#diff-1cb407f308a0912bae7df6e93484f4592fbfdc65ae8e52fa1fc49a5d28e8d700) with the [caseInsensitive = true](https://github.com/antlr/grammars-v4/blob/master/sql/mysql/Positive-Technologies/MySqlLexer.g4#L28) option. This option, described in the [documentation](https://github.com/antlr/antlr4/blob/master/doc/options.md#caseinsensitive), was added in Antlr 4.10 to remove the requirement of implementing a case insensitive stream. The latest version of the Intellij Antlr4 plugin generates using 4.10.1.

I tested this on the newest version of the Intellij Antlr plugin, 1.19.2. I get a parse tree exactly as expect:
<img width="683" alt="tree" src="https://user-images.githubusercontent.com/11389451/173975210-f127569a-185a-48da-9965-e6a3ab7cf961.png">

It works with mixed case.

**Comment by bkiers** [comment]
> > > Don't you need a custom char stream, because the lexer expects upper case keywords?
> 
> > yes,i need a custom char stream,Thank you very much
> 
> Folks, this doesn't make sense.

True. I only read the "Usage, important note" comments from [the repository](https://github.com/antlr/grammars-v4/tree/master/sql/mysql/Positive-Technologies) and later saw the addition of the `options { caseInsensitive = true; }` to the lexer. That's why I removed my comment from this issue. 

The comments are incorrect and ought to be removed (I'll make a PR later on... If I don't forget :)).
